unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Camera/CameraController.cs

```diff
@@ -14,10 +14,13 @@ public class CameraController : MonoBehaviour
         [SerializeField]
         internal new UnityEngine.Camera camera;
 
+        [SerializeField]
+        internal new UnityEngine.Camera skyboxCamera;
+
```

Comment:
Let's move this to the SkyboxPlugin (`SkyboxController`). You will probably need to use the `DataStore.i.camera` and add an `enabled` value (that is toggle here in the `CameraController`) to reacto properly there. The Transform to do the follow is already in the DataStore

Reply:
**The reason for separate script for following the camera**: Camera movement is controlled by Cinemachine which runs after all the process in Unity under Script execution order. So that is why I made a new script for following the camera which is also added in script execution order after the cinemachine.

**Reason for having skyboxCamera inside Camera Controller and prefab**: To include two cameras in URP one have to be the base and another should be Overlay. For present condition skybox camera is acting as the base camera. To make these changes it is required to have skybox camera in the prefab itself, so that if in future someone edit the camera again he will be able to see the dependency. Also the same reason is for having skybox camera in the camera controller script, to make the dependency visible and to have the reference of skybox camera.

Reply:
That's no problem, you can do it programatically using this: https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@7.2/manual/camera-stacking.html
We are already doing some initialization with the main camera in the `SkyboxController`.

Plugins must be self-contained, if we disable the skybox plugin there should be no traces of the skybox feature anywhere.


The script following the camera is fine,  but you can just set the target through code (in the `SkyboxController`) so you dont have hard references in a serialized field and you can decouple the skybox camera from the standard one.

Reply:
A new SkyboxCamera script is there for handling initialization of skybox camera. 
Also, Both the camera are now base cameras with skybox camera as low priority (-5)